src/utils/uploader.js

```diff
@@ -94,25 +105,29 @@ export class BlobUploader {
 
             const hash = commitmentsToVersionedHashes(commitment);
             versionedHashes.push(ethers.hexlify(hash));
-        }
+        });
 
-        // send
-        tx.type = 3;
-        tx.blobVersionedHashes = versionedHashes;
-        tx.blobs = ethersBlobs;
-        tx.kzg = kzg;
-        return await this.#wallet.sendTransaction(tx);
+        await Promise.all(blobPromises);
+        return [ethersBlobs, versionedHashes];
     }
 
```

Comment:
can you document what is the lock for?

Reply:
Blob transactions must be sent sequentially; otherwise, the RPC will crash. However, since blob uploads involve concurrent transactions, a thread lock needs to be implemented at the final sending step to ensure they are sent sequentially.

Reply:
Previously, the thread lock encompassed the entire `sendTxLock` function. Now, the granularity has been reduced so that only the `sendTransaction` step in `ethers` is locked.